mlir/utils/performance/perfRunner.py

```diff
@@ -793,6 +819,133 @@ def benchmarkMIOpenWithMLIRKernels(configs, arch, filename, paths: Paths):
     chip = GFX_CHIP_RE.search(arch).group(0)
     df.to_csv(chip + '_' + filename, index=False)
 
+def findRunCommand(filename, includes=['RUN']):
+    with open(filename, 'r') as f:
+        for line in f:
+            if line.strip().startswith("//") and not 'DISABLE' in line and all(word in line for word in includes):
+                command = line.split("RUN")[1].strip()[2:] # Remove the "//" and "RUN" prefixes and leading/trailing whitespace
+                parts = command.split('|')  # Split the command using the "|" separator
+                return parts[0].strip()
+                # Stop processing lines after finding the first matching line
+    # Not a valid test
+    print("WARNING: cannot find valid RUN command in ", filename)
+    return ""
+
+def getTestVector(file_path, paths:Paths):
+    firstCommand = findRunCommand(file_path)
+    if (not firstCommand):
+        return ""
+
+    if ("-migraphx-to-tosa" in firstCommand):
+        rocmlirOptCommand = [paths.mlir_paths.rocmlir_opt_path, '-migraphx-to-tosa', file_path]
+        rocmlirDriverCommand = [paths.mlir_paths.rocmlir_driver_path, '-host-pipeline', 'partition,highlevel', '-targets', getChip()]
+        # rocmlir-opt -migraphx-to-tosa ../mlir/test/fusion/e2e/resnet50/mixr-resnet-fusion-case-1.mlir
+        p1 = subprocess.Popen(rocmlirOptCommand, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+        # pipe to rocmlir-driver -host-pipeline partition,highlevel -targets gfx90a
+        p2 = subprocess.Popen(rocmlirDriverCommand, stdin=p1.stdout, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+        p1.stdout.close()
+    else:
+        rocmlirDriverCommand = [paths.mlir_paths.rocmlir_driver_path, '-host-pipeline', 'partition,highlevel', '-targets', getChip(), file_path]
+        # rocmlir-driver -host-pipeline partition,highlevel -targets gfx90a
+        p2 = subprocess.Popen(rocmlirDriverCommand, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
+ 
+    # pipe to rocmlir_gen --emit-tuning-key
+    tuningKey = subprocess.Popen([paths.mlir_paths.rocmlir_gen_path, '--emit-tuning-key', '-'], stdin=p2.stdout,
+                          stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    p2.stdout.close()
+    output, _ = tuningKey.communicate()
+    result = output.decode('utf-8').strip().split('\t')
+    return result[1]
+
+def generateProfilingProcesses(file_path, rocmlirGenArgs, paths: Paths):
+    firstCommand = findRunCommand(file_path, ['RUN', 'xmir-runner'])
```

Comment:
Yeah, this might be better parametrized by `hasRunner` as a boolean option and then filtering on the regex output.